### DISCUSSION TITLE
Message data encrypted

### DISCUSSION THREAD

**Discussion by c4r3** [Byron/google-apis-rs#390] [Q&A] [answered]
Hi,
i had a strange issue when i receive new messages in "pull" mode. The data field, the payload of the message, all of a sudden is encrypted (example: Y2lhbyB6aW8=). Few days ago was working like a charm.
I'm testing locally on my laptop with a service account with PubSub Admin role and i have no issues about security, authentication and authorization.
I'm using the last version of the SDK.

Any idea?
Best regards

**Comment by Byron** [comment, marked as answer]
`Y2lhbyB6aW8=` is base64 _encoded_, and I wouldn't know why the server starts doing that or the sender of the message, for that matter.

Please note that this is not the place to get support for how APIs work, this project isn't maintained by Google.

**Reply by c4r3** [reply]
You are right, it's a base 64 encoding. I was capable to read the payload of the message without Base64 decoding (i was thinking the decoding was applyed directly by the SDK) but all of a sudden the data string was returned encoded and some of my unittest failed.
By the way i tested also with the official pubsub web ui with several stub messages.
With the decoding i resolve the issue, thanks for the support.
